src/main/java/com/puppycrawl/tools/checkstyle/filters/PatchXpathFilterElement.java

```diff
@@ -73,26 +78,99 @@ private boolean isFileNameMatching(TreeWalkerAuditEvent event) {
     private boolean isLineMatching(TreeWalkerAuditEvent event) {
         boolean result = false;
         if (event.getLocalizedMessage() != null) {
-            for (List<Integer> singleLineRangeList : lineRangeList) {
-                final int startLine = singleLineRangeList.get(0) + 1;
-                final int endLine = singleLineRangeList.get(1) + 1;
-                final int currentLine = event.getLine();
-                result = currentLine >= startLine && currentLine < endLine;
-                if (result) {
-                    break;
-                }
+            result = lineMatching(event.getLine());
+        }
+        return result;
+    }
+
+    private boolean lineMatching(int currentLine) {
+        boolean result = false;
+        for (List<Integer> singleLineRangeList : lineRangeList) {
+            final int startLine = singleLineRangeList.get(0) + 1;
+            final int endLine = singleLineRangeList.get(1) + 1;
+            result = currentLine >= startLine && currentLine < endLine;
+            if (result) {
+                break;
             }
         }
         return result;
     }
 
     /**
-     * Is matching by xpath query.
+     * Is matching by context strategy.
      *
      * @param event event
      * @return true if it is matching or not set.
      */
-    private boolean isXpathQueryMatching(TreeWalkerAuditEvent event) {
-        return true;
+    private boolean isContextStrategyCheck(TreeWalkerAuditEvent event) {
+        boolean result = false;
+        final String checkShortName = getCheckShortName(event);
+        if ("OuterTypeNumberCheck".equals(checkShortName)) {
+            result = true;
+        }
+        else if ("ReturnCountCheck".equals(checkShortName)
+                || "MethodCountCheck".equals(checkShortName)
+                || "AnonInnerLengthCheck".equals(checkShortName)
+                || "ExecutableStatementCountCheck".equals(checkShortName)
+                || "MethodLengthCheck".equals(checkShortName)) {
```

Comment:
Why do we need to limit check types? Are there TreeWalker checks that will not support this matching algorithm?
If this is still needed, please extract it to a constant list and check it here like `supportedChecks.contains(checkShortName)`.

Reply:
ok, I understand now, some checks will not support context strategy, for example AbstractClassName.
Please move check names to a collection.

Reply:
done